api/src/main/java/com/velocitypowered/api/event/ResultedEvent.java

```diff
@@ -106,4 +107,55 @@ public static ComponentResult denied(@NonNull Component reason) {
             return new ComponentResult(false, reason);
         }
     }
+    /**
+     * Represents an "allowed/allowed with online mode/denied" result with a reason allowed for denial.
+     */
+    class PreLoginComponentResult extends ComponentResult {
```

Comment:
Do not put this in the `ResultedEvent` class. Convention is that if a result is event-specific, it is a subclass of that event.